File: Chess/Models/Entities/Game.cs
using Chess.Enums;

namespace Chess.Models.Entities;

public class Game
{
    public ulong Id { get; init; }
    public GameStatus Status { get; set; } = GameStatus.Pending;
    public GameEnding? Ending { get; init; } = null;
    public PieceColour? Winner { get; init; } = null;
    public TimeControl TimeControl { get; init; }
    public IEnumerable<Move> Moves { get; set; } = new List<Move>();
    public IEnumerable<GamePiece> Pieces { get; set; } = new List<GamePiece>();
    public Guid? PlayerOneId { get; set; } = null;
    public Guid? PlayerTwoId { get; set; } = null;
}

File: Chess/Assets/src/interfaces/Game.ts
import { GameEnding } from "./GameEnding";
import { GameStatus } from "../enums/GameStatus";
import { PieceColour } from "./PieceColour";
import { TimeControl } from "./TimeControl";
import { PieceType } from "./PieceType";
import { PieceStatus } from "./PieceStatus";

interface BasePiece {
  type: PieceType;
  colour: PieceColour;
  initialX: 2;
  initialY: 2;
}

interface GamePiece {
  id: number,
  piece: BasePiece;
  x: number;
  y: number;
  promotionType: PieceType;
  totalMoves: number;
  status: PieceStatus;
}

interface GameMove {
  id: number;
}

export interface Game {
  id: number;
  status: GameStatus;
  ending: GameEnding;
  winner: PieceColour;
  timeControl: TimeControl;

  // TODO: Consolidate this with the other Piece interface
  pieces: GamePiece[];
  moves: GameMove[];

  playerOneId: string;
  playerTwoId: string;
}